src/main/java/com/techsophy/tsf/commons/acl/ACLEvaluator.java

```diff
@@ -0,0 +1,51 @@
+package com.techsophy.tsf.commons.acl;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ACLEvaluator
+{
+    private final String gatewayURL;
+    private final WebClient webClient;
+
+    public ACLEvaluator(String gatewayURL)
+    {
+        this.gatewayURL=gatewayURL;
+        this.webClient=WebClient.builder().build();          //creates only single instance of webClient
+    }
+
+    private ACLValidate evaluateACL(Map<String,?> context,String aclId,String token)
+    {
+        var contextMap= Collections.singletonMap("context",new HashMap<>());
+        var result= webClient
+                .post()
+                .uri(gatewayURL+Constants.getACCOUNTS_ACL_URL() +aclId+"/evaluate")
+                .headers(header -> header.setBearerAuth(token))
+                .headers(header -> header.setContentType(MediaType.APPLICATION_JSON))
+                .bodyValue(contextMap)
+                .retrieve()
+                .bodyToMono(new ParameterizedTypeReference<ResponseBaseModel<ACLValidate>>() {})
+                .blockOptional()
+                .orElseThrow();
+        return result.getData();
+    }
+
+    public ACLDecision getRead(Map<String,?> context,String aclId,String token)
```

Comment:
provide overloaded methods with authorization header.
Would be better if the parameters are reversed, aclId, token and than context.